Custom nuget.exe on newer mono might break due to the MSBuild SDK Resolver

Specifically errors like this can be hit; 

```
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: The \"GetRestoreProjectStyleTask\" task failed unexpectedly. [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/a/a.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: System.MissingMethodException: Method not found: System.Nullable`1<System.ValueTuple`2<NuGet.ProjectModel.ProjectStyle, bool>> NuGet.Commands.MSBuildRestoreUtility.GetProjectRestoreStyle(string,bool,string,string,string,NuGet.Common.ILogger) [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/a/a.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00029] in <10b73e4bf233421cae439ef765e97ef0>:0 [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/a/a.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask (Microsoft.Build.BackEnd.ITaskExecutionHost taskExecutionHost, Microsoft.Build.BackEnd.Logging.TaskLoggingContext taskLoggingContext, Microsoft.Build.BackEnd.TaskHost taskHost, Microsoft.Build.BackEnd.ItemBucket bucket, Microsoft.Build.BackEnd.TaskExecutionMode howToExecuteTask) [0x002a9] in <10b73e4bf233421cae439ef765e97ef0>:0 [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/a/a.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: The \"GetRestoreProjectStyleTask\" task failed unexpectedly. [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/d/d.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: System.MissingMethodException: Method not found: System.Nullable`1<System.ValueTuple`2<NuGet.ProjectModel.ProjectStyle, bool>> NuGet.Commands.MSBuildRestoreUtility.GetProjectRestoreStyle(string,bool,string,string,string,NuGet.Common.ILogger) [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/d/d.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute () [0x00029] in <10b73e4bf233421cae439ef765e97ef0>:0 [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/d/d.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask (Microsoft.Build.BackEnd.ITaskExecutionHost taskExecutionHost, Microsoft.Build.BackEnd.Logging.TaskLoggingContext taskLoggingContext, Microsoft.Build.BackEnd.TaskHost taskHost, Microsoft.Build.BackEnd.ItemBucket bucket, Microsoft.Build.BackEnd.TaskExecutionMode howToExecuteTask) [0x002a9] in <10b73e4bf233421cae439ef765e97ef0>:0 [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/d/d.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: The \"GetRestoreProjectStyleTask\" task failed unexpectedly. [/Users/buildagent/agent/_work/1/s/.test/work/7536c2b5/16d30884/solution/g/g.csproj]
/var/folders/c2/10wzh_315rs5wmnc9m3gsxlw0000gq/T/NuGetScratch/6730wwn7.z40.nugetrestore.targets(437,5): error MSB4018: System.MissingMethodException: Method not found: System.Nullable`1<System.ValueTuple`2<NuGet.ProjectModel.ProjectStyle, bool>> NuGet.Commands.MSBuildRestoreUtility.GetProjectRestoreStyle(string,bool,string,string,string,NuGet.Common.ILogger) [/Users/buildagent/agent/_work/
```

--- a/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs
+++ b/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs
@@ -8,6 +8,7 @@ using System.Collections.Generic;
 using System.Globalization;
 using System.IO;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using Microsoft.Build.Framework;
 
 namespace Microsoft.Build.NuGetSdkResolver
@@ -61,6 +62,7 @@ namespace Microsoft.Build.NuGetSdkResolver
         /// <summary>
         /// Deserializes a global.json and returns the MSBuild SDK versions
         /// </summary>
+        [MethodImpl(MethodImplOptions.NoInlining)]
         private static Dictionary<string, string> Deserialize(string value)
         {
             return JsonConvert.DeserializeObject<GlobalJsonFile>(value).MSBuildSdks;

--- a/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs
+++ b/src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs
@@ -6,6 +6,7 @@ using System.Collections.Generic;
 using System.Globalization;
 using System.IO;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 using Microsoft.Build.Framework;
 using NuGet.Commands;
@@ -193,6 +194,7 @@ namespace Microsoft.Build.NuGetSdkResolver
             /// <summary>
             /// Attempts to parse a string as a NuGetVersion and returns an object containing the instance which can be cast later.
             /// </summary>
+            [MethodImpl(MethodImplOptions.NoInlining)]
             public static bool TryParseNuGetVersion(string version, out object parsed)
             {
                 if (NuGetVersion.TryParse(version, out var nuGetVersion))

